### PR TITLE
Balder/do not allow topk optimisation when distribution is severly skewed take2

### PR DESCRIPTION
**Pull request by baldersheim**
@jonmv PR
Only the last 2 commits are new.

### CODE REVIEW

File: container-search/src/main/java/com/yahoo/search/dispatch/searchcluster/SearchCluster.java [hunk 1]
```diff
@@ -370,7 +370,7 @@ private boolean isGroupNodeCoverageSufficient(int workingNodes, int nodesInGroup
 
     public boolean isGroupWellBalanced(OptionalInt groupId) {
         if (groupId.isEmpty()) return false;
-        Group group = groups().get(groupId);
+        Group group = groups().get(groupId.getAsInt());
```
Comment (jonmv) [inline container-search/src/main/java/com/yahoo/search/dispatch/searchcluster/SearchCluster.java:373]: XD

Comment (baldersheim) [reply container-search/src/main/java/com/yahoo/search/dispatch/searchcluster/SearchCluster.java:373]: ?

Comment (jonmv) [reply container-search/src/main/java/com/yahoo/search/dispatch/searchcluster/SearchCluster.java:373]: Same as 😆

**Review by jonmv — COMMENTED** [review]
Perfect!

**Pull request merged**